### ISSUE TITLE
Calls to methods that are decorated doesn't check signature against correct parameters

### ISSUE THREAD

**Issue by nymann** [microsoft/pyright#2208]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
When calling a method that's decorated by another method Pyright checks arguments against the decorated method and not the decorator method.

**To Reproduce**
```py
import functools

def decorator(func):
    @functools.wraps(func)
    def wrapped(a: str):
        b = int(a)
        return func(b=b)

    return wrapped

def decorated_method(b: int):
    return 2 * b

@decorator
def example(a: str):
    return decorated_method(a=a)

```

**Expected behavior**
No diagnostic message, it should be valid code.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/7005867/130115085-b1e64dd7-2be2-47fe-8328-17711132461d.png)

**VS Code extension or command-line**
Command-line (neovim lsp),
```sh
 $ pyright --version
pyright 1.1.162
```

**Comment by erictraut** [comment]
Did you intend to apply the decorator to `decorated_method`? If so, you need to precede it with `@decorator`. If you do this, the type error will go away.

```python
@decorator
def decorated_method(b: int):
    return 2 * b
```

**Comment by nymann** [comment]
Yeah, I am sleepy it does indeed work in my example code if I do it right.
```py
import functools

def decorator(func):
    @functools.wraps(func)
    def wrapped(a: str):
        b = int(a)
        return func(b=b)

    return wrapped

@decorator
def decorated_method(b: int):
    return 2 * b

def example(a: str):
    return decorated_method(a=a)
```

However, in my "production" code I still have the problem. I'll dig a bit deeper.

**Comment by nymann** [comment]
Feel free to close this issue, I'll open a new one if I can make a minimum viable example that reproduces what I am seeing.
